app/v1/installations/[installationId]/resource-transfer-requests/[claimId]/route.ts

```diff
@@ -0,0 +1,73 @@
+import { NextResponse } from 'next/server';
+import { daleteTransferRequest, getTransferRequest, setTransferRequest } from '@/lib/partner';
+import { Claim, createClaimRequestSchema } from '@/lib/vercel/schemas';
+import { Params } from '../utils';
+import { withAuth } from '@/lib/vercel/auth';
+import { readRequestBodyWithSchema } from '@/lib/utils';
+
+export const PUT = withAuth(
+    async (oidcClaims, request, { params }: { params: Params }) => {
+        const matchingClaim = await getTransferRequest(
+            params.transferId,
+        );
+        
+        if (matchingClaim) {
+            return NextResponse.json({ description: 'Operation failed because of a conflict with the current state of the resource' }, { status: 409 });
+        }
+        
+        const requestBody = await readRequestBodyWithSchema(
+            request,
+            createClaimRequestSchema,
+        );
+
+        const {data} = requestBody;
+        if (!requestBody.success || !data) {
+            return NextResponse.json({ description: 'Input has failed validation' }, { status: 400 });
+        }
+
+        var expirationDate = new Date();
+        expirationDate.setDate(expirationDate.getDate() + 7);
+        const newClaim: Claim = {
+            transferId: params.transferId,
+            status: 'unclaimed',
+            sourceInstallationId: oidcClaims.installation_id, 
+            expiration: data.expiration,
+            resourceIds: data.resourceIds,
+        }
+        await setTransferRequest(newClaim);
+
+        return NextResponse.json({ description: 'Claim created successfully' });
```

Comment:
There shouldn't be any JSON here. The description in the spec is for OpenAPI generation, not for actual use.